mitiq/interface/mitiq_pyquil/conversions.py

```diff
@@ -37,9 +37,11 @@ def to_quil(circuit: Circuit) -> QuilType:
     # if we are using LineQubits, keep the qubit labeling the same
     if isinstance(max_qubit, LineQubit):
         qubit_range = max_qubit.x + 1
-        return circuit.to_quil(qubit_order=LineQubit.range(qubit_range))
+        return str(
+            QuilOutput(circuit.all_operations(), LineQubit.range(qubit_range))
+        )[23:]
     # otherwise, use the default ordering (starting from zero)
-    return circuit.to_quil()
+    return str(QuilOutput(circuit.all_operations(), circuit.all_qubits()))[23:]
```

Comment:
```suggestion
    return str(QuilOutput(circuit.all_operations(), sorted(circuit.all_qubits())))[23:]
```
For some reason I am not sure about, this seems to fix the remaining failing test. At least locally.